restore.py

```diff
@@ -59,6 +59,65 @@ def restoreFromBackup(backup, progress=lambda x: ()):
     except:
         pass
 
+    # should we restore old schema and move to MBR?
+    restore_partitions = None
+    if 'BOOT' not in backup_partition_layout and 'LOG' not in backup_partition_layout:
+        new_restore_partition = restore_partition
+        efi_boot = False
+
+        # remove unneeded partitions
+        if dsk.swap[0]:
+            tool.deletePartition(tool.partitionNumber(dsk.swap[1])) # swap
+        tool.deletePartitionIfPresent(boot_partnum) # boot
+
+        # use logs as root
```

Comment:
It's probably worth sanity checking that the resulting partition is actually the expected size (i.e. root_mbr_size_old).

Reply:
Another option would be to remove all additional partitions earlier, create new root with expected size. So another round of repartitioning. I'll think more about

Reply:
Checked size against `constants.root_gpt_size_old`